src/move/econia/sources/market.move

```diff
@@ -3089,13 +3238,31 @@ module econia::market {
             size = if (still_crosses_spread || self_match_cancel) 0 else
                 // Else update size to amount left to fill post-match.
                 size - (base_traded / order_book_ref_mut.lot_size);
-        }; // Done with optional matching as a taker across the spread.
-        // Return without market order ID if immediate-or-cancel or if
-        // remaining size to fill after matching does not meet minimum
-        // size requirement for market.
+            // Extract fill event queue packed by matching engine.
+            option::extract(&mut fill_event_queue_option)
+        } else { // If spread not crossed (matching engine not called):
+            // Increment order counter.
+            order_book_ref_mut.counter = order_book_ref_mut.counter + 1;
```

Comment:
Comment simply repeats code

Reply:
Tagging #321 to incorporate refactors on that iteration